src/indra_cogex/Analysis/gene_blueprint.py

```diff
@@ -0,0 +1,206 @@
+"""Gene-centric analysis blueprint."""
+
+from typing import Dict, List, Mapping, Tuple
+import pandas as pd
+from indra.databases import hgnc_client
+from indra_cogex.client.enrichment.continuous import (
+    get_human_scores,
+    get_mouse_scores,
+    get_rat_scores,
+    indra_downstream_gsea,
+    indra_upstream_gsea,
+    phenotype_gsea,
+    reactome_gsea,
+    wikipathways_gsea,
+    go_gsea
+)
+
+from indra_cogex.client.enrichment.discrete import (
+    go_ora,
+    indra_downstream_ora,
+    indra_upstream_ora,
+    phenotype_ora,
+    reactome_ora,
+    wikipathways_ora,
+)
+
+from ...client.enrichment.signed import reverse_casual_reasoning
+
+
+def parse_genes_field(s: str) -> Tuple[Dict[str, str], List[str]]:
```

Comment:
This function seems to relate to processing the form fields. In my opinion, this is something related to the input from the web rather than the analysis, and therefore this function could be kept in `indra_cogex/apps/gla/gene_blueprint.py` and the parsed gene list coming out from this function could be passed on to the various functions in here. The functions in here and in `indra_cogex/Analysis/metabolite_blueprint.py` should assume that the provided genes or metabolites are already parsed.